File: 0293. Flip Game/Solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    vector<string> generatePossibleNextMoves(string currentState)
    {
        vector<string> res;
        for (size_t i = 1; i < currentState.size(); i++) {
            if (currentState[i - 1] == '+' && currentState[i] == '+') {
                res.emplace_back(currentState);
                res.back()[i - 1] = res.back()[i] = '-';
            }
        }
        return res;
    }
};

File: 0293. Flip Game/Solution.py
from typing import List


class Solution:
    def generatePossibleNextMoves(self, currentState: str) -> List[str]:
        res = []
        for i in range(1, len(currentState)):
            if currentState[i - 1] == currentState[i] == "+":
                res.append(currentState[:i-1] + "--" + currentState[i+1:])
        return res
